core/templates/app-events/app-events.ts

```diff
@@ -19,7 +19,7 @@
 // The base-class for all of the events that the application pushes onto the
 // MessageQueue. The only guarantee that this class makes is a read-only Type.
 export abstract class BaseEvent {
-  public readonly type: string;
+  public readonly type: string = '';
```

Comment:
I'm not sure of the property usage here, adding `| undefined` as type could also solve the issue,why do we need to initialize it with an empty string?

Reply:
yes, you are right... we shouldn't initialize it with anything actually. This value is initialized inside the Event classes which extend this baseEvent class. Like stated above, the only purpose of this class is to ensure that `Type` exists.